lib/view_component/base.rb

```diff
@@ -91,14 +91,21 @@ def render_in(view_context, &block)
       before_render
 
       if render?
-        render_template_for(@variant)
+        render_template_for(@variant) + _after_render
       else
         ""
       end
     ensure
       @current_template = old_current_template
     end
 
+    # EXPERIMENTAL: Optional content to be returned after the rendered template.
+    #
+    # @return [String]
+    def _after_render
```

Comment:
Thinking a bit more broadly about this callback, I'd love to start chatting about/looking into compiler level hooks instead of relying on component level hooks.

Ignoring that and focusing on naming, thoughts on something like the following:

```suggestion
    def _output_postamble
```

`_after_render` sounds like a callback since we have `before_render` and not something that will impact the output of the component.